src/objects.rs

```diff
@@ -41,27 +42,70 @@ impl ObjectData {
     pub(crate) fn new(
         parser: &mut impl Iterator<Item = XmlEventResult>,
         attrs: Vec<OwnedAttribute>,
-        tilesets: Option<&[MapTilesetGid]>,
+        tilesets: &[MapTilesetGid],
+        templates: &mut Vec<Template>,
+        for_template: Option<usize>,
+        base_path: Option<&Path>,
+        cache: &mut impl ResourceCache,
     ) -> Result<ObjectData, TiledError> {
-        let ((id, tile, n, t, w, h, v, r), (x, y)) = get_attrs!(
+        let (
+            (mut id, mut tile, mut x, mut y, mut n, mut t, mut w, mut h, mut v, mut r, template),
+            (),
+        ) = get_attrs!(
             attrs,
             optionals: [
                 ("id", id, |v:String| v.parse().ok()),
                 ("gid", tile, |v:String| v.parse().ok()
-                                            .and_then(|bits| LayerTileData::from_bits(bits, tilesets?))),
+                                            .and_then(|bits| LayerTileData::from_bits(bits, tilesets, for_template))),
+                ("x", x, |v:String| v.parse().ok()),
+                ("y", y, |v:String| v.parse().ok()),
                 ("name", name, |v:String| v.parse().ok()),
                 ("type", obj_type, |v:String| v.parse().ok()),
                 ("width", width, |v:String| v.parse().ok()),
                 ("height", height, |v:String| v.parse().ok()),
                 ("visible", visible, |v:String| v.parse().ok().map(|x:i32| x == 1)),
                 ("rotation", rotation, |v:String| v.parse().ok()),
+                ("template", template, |v:String| v.parse().ok()),
             ],
-            required: [
-                ("x", x, |v:String| v.parse().ok()),
-                ("y", y, |v:String| v.parse().ok()),
-            ],
+            required: [],
             TiledError::MalformedAttributes("objects must have an x and a y number".to_string())
         );
+
+        // If the template attribute is there, we need to go fetch the template file
+        let template_id = if let Some(template) = template {
+            let s: String = template;
+            let parent_dir = base_path
+                .ok_or(TiledError::SourceRequired {
+                    object_to_parse: "Template".to_string(),
+                })?
+                .parent()
+                .unwrap();
```

Comment:
This should return an error instead; I believe there was an `InvalidPath` error or similar in the interface.